sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90

```diff
@@ -42,69 +44,72 @@ module pmat
 
 contains
 
-!=============================================================================
+!> Swap vectors
+!!
+!! @param d vector
+!! @param e vector
+!! @author R. J. Purser
 subroutine sswpvv(d,e)!                                                [swpvv]
-!=============================================================================
-! Swap vectors
-!-------------
 real(sp),    intent(inout) :: d(:), e(:)
 real(sp)                   :: tv(size(d))
-!=============================================================================
 tv = d; d = e; e = tv
 end subroutine sswpvv
-!=============================================================================
+
+!> dswpvv routine
+!! @author R. J. Purser
 subroutine dswpvv(d,e)!                                                [swpvv]
-!=============================================================================
 real(dp), intent(inout) :: d(:), e(:)
 real(dp)                :: tv(size(d))
-!=============================================================================
 tv = d; d = e; e = tv
 end subroutine dswpvv
-!=============================================================================
+
+!> cswpvv routine
+!! @author R. J. Purser
 subroutine cswpvv(d,e)!                                                [swpvv]
-!=============================================================================
 complex(dpc),intent(inout) :: d(:), e(:)
 complex(dpc)               :: tv(size(d))
-!=============================================================================
 tv = d; d = e; e = tv
 end subroutine cswpvv
 
-!=============================================================================
+!> sinvmt routine
+!! @author R. J. Purser
 subroutine sinvmt(a)!                                                    [inv]
-!=============================================================================
 real(sp),dimension(:,:),intent(INOUT):: a
 logical                              :: ff
 call sinvmtf(a,ff)
 if(ff)stop 'In sinvmt; Unable to invert matrix'
 end subroutine sinvmt
-!=============================================================================
+
+!> dinvmt routine
```

Comment:
Always capitalize and punctuate, especially first sentence. This has been mentioned many times.

Reply:
We normally direct copy the comment from original author's code remark. Looked at the record, can't see specified many times.